client/src/components/ColorList.js

```diff
@@ -1,42 +1,95 @@
 import React, { useState } from "react";
-import axios from "axios";
+// import axios from "axios";
+import { axiosWithAuth } from "../auth/axiosWithAuth";
+import uuid from "uuid"
 
 const initialColor = {
   color: "",
   code: { hex: "" }
 };
 
-const ColorList = ({ colors, updateColors }) => {
+const ColorList = ({ colors, updateColors, fetchColors }) => {
   console.log(colors);
   const [editing, setEditing] = useState(false);
   const [colorToEdit, setColorToEdit] = useState(initialColor);
+  const [newColor, setNewColor] = useState({
+    color:"",
+    code:{hex:""},
+    id:uuid() 
+  });
 
   const editColor = color => {
     setEditing(true);
     setColorToEdit(color);
   };
 
+
   const saveEdit = e => {
     e.preventDefault();
+    console.log(colorToEdit);
     // Make a put request to save your updated color
     // think about where will you get the id from...
     // where is is saved right now?
+    axiosWithAuth()
+    .put(`http://localhost:5000/api/colors/${colorToEdit.id}`, colorToEdit)
+    .then(res => {
+      setEditing(false);
+      setColorToEdit(initialColor);
+      fetchColors();
+    })
+    .catch(error => {
+      debugger;
+      console.error(error);
+    });
+  };
+
+ 
+  const handleSubmitNewColor = e => {
+    e.preventDefault();
+    console.log(e.target.name);
+ 
+    axiosWithAuth()
+    .post(`http://localhost:5000/api/colors`, newColor)
+    .then(res => {
+            updateColors(
```

Comment:
Lastly, `updateColors` is the setter function for the array of colors that `BubbleList` component is mapping over.  You made the mistake of resetting to a color object rather than to an array that a successful `res.data` will return. I think you meant to use `setNewColor` to reset the form

```
.then(res => {
        setNewColor({
          ...newColor,
          color: '',
          code: { hex: '' }
        });
        updateColors(res.data);
      })
```